### PR TITLE
Fix Database.load() for db with misc table as scheme

### LINKED ISSUE #294: Loading a database with misc table might fail
The following example creates a databases with a misc table as scheme

```python
import audformat.testing
import pandas as pd

db = audformat.testing.create_db(minimal=True)
db.schemes['scheme'] = audformat.Scheme(
    labels=['some', 'test', 'labels']
)   
audformat.testing.add_misc_table(
    db,
    'misc-in-scheme',
    pd.Index([0, 1, 2], dtype='Int64', name='idx'),
    columns={'emotion': ('scheme', None)}
)
db.schemes['misc'] = audformat.Scheme(
    'int',
    labels='misc-in-scheme',
)
db.save('./db-test')
```

When we then try to load the database with
```python
db = audformat.Database.load('./db-test')
```
we get
```
---------------------------------------------------------------------------
BadKeyError                               Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 db = audformat.Database.load('./db-test')

File ~/git/audeering/audformat/audformat/core/database.py:975, in Database.load(root, name, load_data, num_workers, verbose)
    972 with open(path, 'r') as fp:
    974     header = yaml.load(fp, Loader=Loader)
--> 975     db = Database.load_header_from_yaml(header)
    977     params = []
    978     table_ids = []

File ~/git/audeering/audformat/audformat/core/database.py:1072, in Database.load_header_from_yaml(header)
   1070         scheme = Scheme()
   1071         scheme.from_dict(scheme_d)
-> 1072         db.schemes[scheme_id] = scheme
   1074 if 'splits' in header and header['splits']:
   1075     for split_id, split_d in header['splits'].items():

File ~/git/audeering/audformat/audformat/core/common.py:74, in HeaderDict.__setitem__(self, key, value)
     72 def __setitem__(self, key, value):
     73     if self.set_callback is not None:
---> 74         value = self.set_callback(key, value)
     75     if self.value_type is not None:
     76         if not isinstance(value, self.value_type):

File ~/git/audeering/audformat/audformat/core/database.py:1119, in Database._set_scheme(self, scheme_id, scheme)
   1117 scheme._id = scheme_id
   1118 if hasattr(scheme, 'labels') and scheme.labels is not None:
-> 1119     scheme._check_labels(scheme.labels)
   1120 return scheme

File ~/git/audeering/audformat/audformat/core/scheme.py:397, in Scheme._check_labels(self, labels)
    395 for column in self._db.misc_tables[table_id].columns.values():
    396     if column.scheme_id is not None:
--> 397         scheme = self._db.schemes[column.scheme_id]
    398         if scheme.uses_table:
    399             raise ValueError(
    400                 f"The misc table "
    401                 f"'{table_id}' "
   (...)
    405                 f"uses labels from a misc table."
    406             )

File ~/git/audeering/audformat/audformat/core/common.py:82, in HeaderDict.__getitem__(self, key)
     80 def __getitem__(self, key):
     81     if key not in self:
---> 82         raise BadKeyError(key, list(self))
     83     value = super().__getitem__(key)
     84     if self.get_callback is not None:

BadKeyError: "Bad key 'scheme', expected one of []"
```

### PR DESCRIPTION
**Pull request by hagenw**
Closes #294 

Ensures to first assign schemes that are not using a misc table to the database during loading as `Scheme._check_labels()` need those assigned to the database when checking schemes of a misc table used as a scheme.

### CODE REVIEW

**Comment by hagenw** [comment]
There is one downside of the current fix: before `list(db.schemes)` returned a sorted list of schemes, e.g. for `emodb` you would get:
```
['age', 'confidence', 'duration', 'emotion', 'gender', 'language', 'speaker', 'transcription']
```
now schemes using misc tables would be moved to the end of this list:
```
['age', 'confidence', 'duration', 'emotion', 'gender', 'language', 'transcription', 'speaker']
```

**Comment by hagenw** [comment]
I now found a way to resort the `db.schemes` entries at the end. I also added a test to check the expected scheme order which fails without the sorting.

**Comment by hagenw** [comment]
I just realized the order should be given by the order of assignment and not by sorting, so I removed the sorting again.

I need to check why the results returned in `audb` are now different than before.

**Comment by hagenw** [comment]
It took a while, but now I have figured out the problem of order of scheme IDs:

* when schemes are added to a database, `list(db.schemes)` should return them in the order they were added
* when a database is loaded from disk, `list(db.schemes)` should be alphabetically ordered

I adjusted the tests and the code.

This is now ready for review.

**Comment by frankenjoe** [comment]
> * when schemes are added to a database, `list(db.schemes)` should return them in the order they were added
> 
> * when a database is loaded from disk, `list(db.schemes)` should be alphabetically ordered

Ok, so now the order is different after loading a database, right? Would it maybe make more sense to always keep the list sorted?

**Comment by hagenw** [comment]
> > * when schemes are added to a database, `list(db.schemes)` should return them in the order they were added
> > * when a database is loaded from disk, `list(db.schemes)` should be alphabetically ordered
> 
> Ok, so now the order is different after loading a database, right? Would it maybe make more sense to always keep the list sorted alphabetically?

No, the order is the same as before (`master`). We always ordered alphabetically when loading from disk.

When adding stuff to a database, I think it makes sense to present it in the order it was added as a user might want to remove the last entry. If you disagree with this, open a new issue as it would break the current behavior.

File: audformat/core/database.py [hunk 1]
```diff
@@ -1066,10 +1066,30 @@ def load_header_from_yaml(header: dict) -> 'Database':
                 db.raters[rater_id] = rater
 
         if 'schemes' in header and header['schemes']:
+            misc_table_schemes = {}
             for scheme_id, scheme_d in header['schemes'].items():
+                # ensure to load first all non misc table schemes
+                # as they might be needed
+                # when checking the scheme
```
[the remark this answers is not in the document]
Comment (frankenjoe) [reply audformat/core/database.py:1073]: yes or simply "column schemes"?

**Comment by frankenjoe** [comment]
> **Update**: I guess I misunderstood "after". Yes the order can be different after loading a database.

There is a `'sorted_iter'

> When adding stuff to a database, I think it makes sense to present it in the order it was added as a user might want to remove the last entry. If you disagree with this, open a new issue as it would break the current behavior.

I created https://github.com/audeering/audformat/issues/296

File: audformat/core/database.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hagenw) [reply audformat/core/database.py]: Changed accordingly.

**Comment by frankenjoe** [comment]
Cool, good we fixed this bug!

**Pull request merged**